components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/LabelsApiServiceImpl.java

```diff
@@ -53,11 +53,12 @@
     public Response labelsLabelIdDelete(String labelId, String ifMatch, String ifUnmodifiedSince) {
         try {
             APIAdmin apiAdmin = new APIAdminImpl();
-            apiAdmin.deleteLabel(labelId);
+            String user = RestApiUtil.getLoggedInUsername();
+            apiAdmin.deleteLabel(user, labelId);
             return Response.ok().build();
         } catch (APIManagementException e) {
             String errorMessage = "Error while deleting API : " + labelId;
-            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+            RestApiUtil.handleInternalServerError(errorMessage, log);
```

Comment:
where do we log the error trace now?

Reply:
we are logging 
APIUtil.handleException("Unable to delete label. Label is already attached to a API");

Do we need error trace anymore?

Reply:
where do we log that?

Reply:
in org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java